homeassistant/components/samsungtv/media_player.py

```diff
@@ -6,6 +6,7 @@
 from samsungctl import exceptions as samsung_exceptions, Remote as SamsungRemote
 import voluptuous as vol
 import wakeonlan
+from websocket._exceptions import WebSocketException
```

Comment:
Trying to understand and connect the dots.

In the trace, we raised `websocket._exceptions.WebSocketTimeoutException`

I presume here that `WebSocketException` is the base exception, and `WebSocketTimeoutException` is derived from that?

Yes - this is the case, confirmed in `/srv/homeassistant/lib/python3.6/site-packages/websocket/_exceptions.py`.

Ah-ha, found it upstream too;

https://github.com/websocket-client/websocket-client/blob/master/websocket/_exceptions.py#L58

Reply:
``` python
class WebSocketTimeoutException(WebSocketException):
    """
    WebSocketTimeoutException will be raised at socket timeout during read/write data.
    """
```

Reply:
The exceptions are available from the base package:
https://github.com/websocket-client/websocket-client/blob/3c25814664fef5b78716ed8841123ed1c0d17824/websocket/__init__.py#L25